sdk/src/lib/ik-video/ik-video.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, OnInit, ElementRef, Input } from '@angular/core';
+import { ImagekitService } from '../imagekit-angular.service';
+import { IkVideoComponentOptions, QueryParameters, Dict } from '../utility/ik-type-def-collection';
+import { buildSrc, SrcOptions } from '@imagekit/javascript';
+import type { Transformation } from '@imagekit/javascript'
+
+
+@Component({
+  selector: 'ik-video',
+  template: `<video></video>`,
+})
+
+export class IkVideoComponent implements OnInit {
+  @Input('urlEndpoint') urlEndpoint: string;
+  @Input('src') src: string = "";
+  @Input('transformation') transformation: Array<Transformation> = [];
+  @Input('transformationPosition') transformationPosition: "path" | "query";
+  @Input('queryParameters') queryParameters: QueryParameters;
+  url = '';
+
+  constructor(private el: ElementRef, private imagekit: ImagekitService) { }
+
+  ngOnInit(): void {
+    const options: IkVideoComponentOptions =  {
+      src: this.src,
+      urlEndpoint: this.urlEndpoint ? this.urlEndpoint : this.imagekit._ikInstance.options.urlEndpoint,
+      transformation: this.transformation,
+      transformationPosition: this.transformationPosition,
+      queryParameters: this.queryParameters,
+    };
+    this.setUrl(options);
+  }
+  ngOnChanges(): void {
```

Comment:
Calling lifecycle hooks manually is anti-pattern. Instead, extract the logic in these hooks into separate functions and then call them from both the places.

Ref: https://www.linkedin.com/pulse/why-you-shouldnt-trigger-lifecycle-hooks-manually-prakhar-srivastava-ybxnf/